### PR TITLE
fix: setDepartment Livechat API call overriding triggers

### PR DESCRIPTION
**Pull request by MartinSchoeler**
CORE-50

REQUIRES https://github.com/RocketChat/Rocket.Chat/pull/31268

### CODE REVIEW

File: apps/meteor/packages/rocketchat-livechat/assets/demo.html [hunk 1]
```diff
@@ -31,15 +31,6 @@ <h2>without changing page title</h2>
...
 	<a href="#page-7">page 7</a><br>
-
-	<script type="text/javascript">
-	(function(w, d, s, u) {
-		w.RocketChat = function(c) { w.RocketChat._.push(c) }; w.RocketChat._ = []; w.RocketChat.url = u;
-		var h = d.getElementsByTagName(s)[0], j = d.createElement(s);
-		j.async = true; j.src = `${window.location.origin}/livechat/rocketchat-livechat.min.js?_=201903270000`;
-		h.parentNode.insertBefore(j, h);
-	})(window, document, 'script', `${window.location.origin}/livechat`);
-	</script>
```
Comment (MarcosSpessatto) [inline apps/meteor/packages/rocketchat-livechat/assets/demo.html:42]: Why are we removing this?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/packages/rocketchat-livechat/assets/demo.html (hunk 1 above)
Comment (MartinSchoeler) [reply apps/meteor/packages/rocketchat-livechat/assets/demo.html]: It is a fix that I have to do to the demo page, we have a duplicated livechat install there, but I'll do it on a dedicated PR

File: packages/livechat/src/routes/Register/index.tsx [hunk 2]
```diff
@@ -96,8 +97,9 @@ export const Register: FunctionalComponent<{ path: string }> = () => {
 	};
 
 	const getDepartmentDefault = () => {
-		if (departments.some((dept) => dept._id === guestDepartment)) {
-			return guestDepartment;
+		const dept = departments.find((dept) => dept._id === defaultDepartment || dept.name === defaultDepartment);
```
[the remark this answers is not in the document]
Comment (MartinSchoeler) [reply packages/livechat/src/routes/Register/index.tsx:101]: Yep, I kept the same behavior of the setDepartment hook

https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/livechat/src/lib/hooks.ts#L95

[the remark this answers is not in the document]
Comment (MarcosSpessatto) [reply packages/livechat/src/routes/Register/index.tsx:101]: I was about to ask that, thanks @MartinSchoeler! 😅

File: .changeset/afraid-mugs-priest.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"@rocket.chat/meteor": patch
+"@rocket.chat/core-typings": patch
+"@rocket.chat/livechat": patch
+---
+
+fix: setDepartment Livechat API call overriding triggers
```
Comment (MarcosSpessatto) [inline .changeset/afraid-mugs-priest.md:7]: Can we explain it more?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by KevLehman — APPROVED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Pull request merged**